### PR TITLE
Update extract and explode bands test

### PR DESCRIPTION
**Pull request by jgersak**
+ fixture change from test_eopatch -> patch
+ add types
+ split test that test failing

### CODE REVIEW

File: core/eolearn/tests/test_core_tasks.py [hunk 1]
```diff
@@ -379,47 +379,48 @@ def test_map_features(test_eopatch):
...
 )
 def test_explode_bands(
-    test_eopatch: EOPatch,
-    feature: FeatureType,
+    patch: EOPatch,
+    feature: Tuple[FeatureType, str],
     task_input: Dict[Tuple[FeatureType, str], Union[int, Iterable[int]]],
-):
+) -> None:
     move_bands = ExplodeBandsTask(feature, task_input)
```
Comment (zigaLuksic) [inline core/eolearn/tests/test_core_tasks.py:398]: the 'move' naming is weird and doesn't fit this test (or the 'extract bands' test)

File: core/eolearn/tests/test_core_tasks.py [hunk 2]
```diff
@@ -379,47 +379,48 @@ def test_map_features(test_eopatch):
...
+    bands = [2, 4, 6]
+    move_bands = ExtractBandsTask((FeatureType.DATA, "bands"), (FeatureType.DATA, "MOVED_BANDS"), bands)
+    patch = move_bands(patch)
+    assert np.array_equal(patch.data["MOVED_BANDS"], patch.data["bands"][..., bands])
 
     old_value = patch.data["MOVED_BANDS"][0, 0, 0, 0]
     patch.data["MOVED_BANDS"][0, 0, 0, 0] += 1.0
-    assert patch.data["REFERENCE_SCENES"][0, 0, 0, bands[0]] == old_value
+    assert patch.data["bands"][0, 0, 0, bands[0]] == old_value
     assert old_value + 1.0 == approx(patch.data["MOVED_BANDS"][0, 0, 0, 0])
```
Comment (zigaLuksic) [inline core/eolearn/tests/test_core_tasks.py:417]: this is a weird and unclear test. Figure out what it does and replace it with something better

File: core/eolearn/tests/test_core_tasks.py [hunk 3]
```diff
@@ -379,47 +379,48 @@ def test_map_features(test_eopatch):
...
 
-    bands = [2, 4, 16]
-    move_bands = ExtractBandsTask((FeatureType.DATA, "REFERENCE_SCENES"), (FeatureType.DATA, "MOVED_BANDS"), bands)
+
+def test_extract_bands_fails(patch: EOPatch) -> None:
```
Comment (zigaLuksic) [inline core/eolearn/tests/test_core_tasks.py:420]: do we have a custom error for this? or why is this tested?

Comment (jgersak) [reply core/eolearn/tests/test_core_tasks.py:420]: define custom we have `ValueError` if not all bands are smaller then number of featuers but its not custom like `EOErrorLenght` 

from blame Matej added it some time ago, and I did not find good reason to remove or not it 
(aka i m not against removing it)

Comment (zigaLuksic) [reply core/eolearn/tests/test_core_tasks.py:420]: is the error raised by us, or is it just an error raised by `numpy` due to indexing?

Comment (jgersak) [reply core/eolearn/tests/test_core_tasks.py:420]: Error is raised by us.

Comment (zigaLuksic) [reply core/eolearn/tests/test_core_tasks.py:420]: ok, then the test stays

File: core/eolearn/tests/test_core_tasks.py (hunk 1 above)
Comment (jgersak) [reply core/eolearn/tests/test_core_tasks.py]: Updated, name not needed

File: core/eolearn/tests/test_core_tasks.py (hunk 2 above)
Comment (jgersak) [reply core/eolearn/tests/test_core_tasks.py]: It checks that `patch.data["bands"]` and `patch.data["new"]` are two different object replaced with id test

File: core/eolearn/tests/test_core_tasks.py [hunk 4]
```diff
@@ -406,8 +406,9 @@ def test_explode_bands(
 def test_extract_bands(patch: EOPatch) -> None:
     bands = [2, 4, 6]
     patch = ExtractBandsTask((FeatureType.DATA, "bands"), (FeatureType.DATA, "MOVED_BANDS"), bands)(patch)
+
     assert_array_equal(patch.data["MOVED_BANDS"], patch.data["bands"][..., bands])
-    assert id(patch.data["MOVED_BANDS"]) != id(patch.data["bands"])
+    assert all([(id(patch.data["MOVED_BANDS"][band]) != id(patch.data["bands"][band]) for band in bands)])
```
Comment (zigaLuksic) [inline core/eolearn/tests/test_core_tasks.py:411]: I see you're very enthusiastic about `id` tests, but you're underestimating numpy.
Numpy can also return so called `views`. Look at the following snippet from the console:

```python
In [1]: import numpy as np

In [2]: x = np.ones((10, 10))

In [3]: y = x[3:5, 3:5]

In [4]: z = x[3:5, 3:5]

In [5]: id(y) == id(z)
Out[5]: False

In [6]: z[0, 0] = 0

In [7]: y
Out[7]: 
array([[0., 1.],
       [1., 1.]])
```

File: core/eolearn/tests/test_core_tasks.py (hunk 1 above)
Comment (zigaLuksic) [reply core/eolearn/tests/test_core_tasks.py]: you still use the `move` naming, e.g. `"MOVED_BANDS"`

Comment (jgersak) [reply core/eolearn/tests/test_core_tasks.py]: Renamed

**Pull request merged**